lib/AST/DiagnosticEngine.cpp

```diff
@@ -577,7 +577,21 @@ static void formatDiagnosticArgument(StringRef Modifier,
       if (shouldShowAKA(type, typeName)) {
```

Comment:
shouldShowAKA() has this case, presumably to avoid this problem in other contexts:

```
  // Don't show generic type parameters.
  if (type->getCanonicalType()->hasTypeParameter())
    return false;
```

(Note that GenericFunctionTypes respond with false to `hasTypeParameter()`, since the type parameters are conceptually "bound" and not "free")

Maybe we don't need this check at all anymore?

Reply:
Ah yeah, maybe this should become the outer test for using desugaring transform in the sense that if we have type parameters or a generic function types, let's avoid canonicalization and just desugar instead. But if desugar transform can work for all types this could simply be gone, I'll check it :)

Reply:
With the new approach this is indeed not needed!